crates/wit-bindgen-rust/src/interface.rs

```diff
@@ -316,42 +305,148 @@ fn {method_name}(
+                // or provide another way for them to be accessed
+                FunctionKind::Static(id) => {
+                    let resource_name = self.resolve.types[*id]
+                        .name
+                        .as_ref()
+                        .unwrap()
+                        .to_upper_camel_case();
+                    let method_name = name
+                        .to_snake_case()
+                        .replace("static_", "")
+                        .replace(&format!("{}_", resource_name.to_snake_case()), "");
+
+                    uwrite!(
+                        self.src,
+                        "match <T as Handler<Ctx>>::{resource_name}::{method_name}(context, ",
+                        method_name = to_rust_ident(&method_name)
+                    );
+                }
+            }
+
+            // print function arguments
             for i in 0..params.len() {
+                // skip self
+                if i == 0 && matches!(kind, FunctionKind::Method(_)) {
+                    continue;
+                }
+
                 self.src.push_str("params.");
                 self.src.push_str(&i.to_string());
                 self.src.push_str(",");
             }
-            uwrite!(
-                self.src,
-                r#").await {{
-                    Ok(res) => {{
-                        if let Err(err) = transmitter.transmit_static(result_subject, res).await {{
-                            {tracing}::error!(?err, "failed to transmit result")
+
+            // handle function call return value
+            match kind {
+                FunctionKind::Constructor(id) => {
+                    let resource_name = self.resolve.types[*id]
+                        .name
+                        .as_ref()
+                        .unwrap()
+                        .to_upper_camel_case();
+
+                    uwrite!(
+                        self.src,
+                        r#").await {{
+                            Ok(res) => {{
+                                let resource = {resource_name}::new();
+
+                                let resource_table_ref = handler.table();
+                                let mut resource_table = resource_table_ref.lock().await;
+                                resource_table.insert(resource.id.clone(), Box::new(res));
+
+                                if let Err(err) = transmitter.transmit_static(result_subject, resource).await {{
+                                    {tracing}::error!(?err, "failed to transmit result")
+                                }}
+                            }}
+                            Err(err) => {{
+                                if let Err(err) = transmitter.transmit_static(error_subject, err).await {{
+                                    {tracing}::error!(?err, "failed to transmit error")
+                                }}
+                            }}
```

Comment:
This will also need to start serving all the methods on the resource using the ID as the "instance", that's how it works in Go bindgen https://github.com/wrpc/wrpc/blob/a75b04391524217058fb4c5284aee224ee8e4464/crates/wit-bindgen-go/src/interface.rs#L2927-L3165 (I'd suggest checking the code generated in `tests/go/bindings/resources_server/exports/wrpc_test/integration/resources /bindings.wrpc.go)